tap_xero/client.py

```diff
@@ -109,9 +200,52 @@ def filter(self, tap_stream_id, since=None, **params):
 
         request = requests.Request("GET", url, headers=headers, params=params)
         response = self.session.send(request.prepare())
-        response.raise_for_status()
-        response_meta = json.loads(response.text,
-                                   object_hook=_json_load_object_hook,
-                                   parse_float=decimal.Decimal)
-        response_body = response_meta.pop(xero_resource_name)
-        return response_body
+
+        if response.status_code != 200:
+            raise_for_error(response)
+            return None
+        else:
+            response_meta = json.loads(response.text,
+                                    object_hook=_json_load_object_hook,
+                                    parse_float=decimal.Decimal)
+            response_body = response_meta.pop(xero_resource_name)
+            return response_body
+
+
+def raise_for_error(resp):
+    try:
+        resp.raise_for_status()
+    except (requests.HTTPError, requests.ConnectionError) as error:
+        try:
+            error_code = resp.status_code
+
+            # Handling status code 429 specially since the required information is present in the headers
+            if error_code == 429:
+                resp_headers = resp.headers
+                api_rate_limit_message = ERROR_CODE_EXCEPTION_MAPPING[429]["message"]
+                message = "HTTP-error-code: 429, Error: {}. Please retry after {} seconds".format(api_rate_limit_message, resp_headers.get("Retry-After"))
+            # Handling status code 403 specially since response of API does not contain enough information
+            elif error_code in (403, 401):
```

Comment:
Code added w.r.to TDL-648 (Permission Denied error should throw a user friendly message)